acceptance: TestCompose failed [port 26257 failed: Connection refused]

acceptance.TestCompose [failed](https://mesolite.cluster.engflow.com/invocations/default/fa51a298-977d-4dda-9004-d4d515457c95?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvYWNjZXB0YW5jZTphY2NlcHRhbmNlX3Rlc3Q=) on master @ [779a69bf2ae2219e7705dc892256d43725698963](https://github.com/cockroachdb/cockroach/commits/779a69bf2ae2219e7705dc892256d43725698963):


```
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun  3 20:04 ca.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  3 20:04 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun  3 20:04 client.root.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  3 20:04 client.root.key
        psql_1       | -rw------- 1 roach roach 1217 Jun  3 20:04 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2477 Jun  3 20:04 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun  3 20:04 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  3 20:04 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1216 Jun  3 20:04 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun  3 20:04 node.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  3 20:04 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=8111b18f833a
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.16.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        cockroach_1  | *
        cockroach_1  | * WARNING: Running a server without --sql-addr, with a combined RPC/SQL listener, is deprecated.
        cockroach_1  | * This feature will be removed in a later version of CockroachDB.
        cockroach_1  | *
        Aborting on container exit...
        
�[Kpsql_1 exited with code 2
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (43.14s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39192

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15501810?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15501810?buildTab=artifacts#/) on master @ [d5f8472453f71e9fc742970e4e1f2ed96661cc46](https://github.com/cockroachdb/cockroach/commits/d5f8472453f71e9fc742970e4e1f2ed96661cc46):


```
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun  3 20:55 ca.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  3 20:55 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun  3 20:55 client.root.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun  3 20:55 client.root.key
        psql_1       | -rw------- 1 roach roach 1218 Jun  3 20:55 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2611 Jun  3 20:55 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun  3 20:55 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  3 20:55 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1216 Jun  3 20:55 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun  3 20:55 node.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun  3 20:55 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=80eaf353bac3
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.32.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        cockroach_1  | *
        cockroach_1  | * WARNING: Running a server without --sql-addr, with a combined RPC/SQL listener, is deprecated.
        cockroach_1  | * This feature will be removed in a later version of CockroachDB.
        cockroach_1  | *
        Aborting on container exit...
        
�[Kpsql_1 exited with code 2
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (32.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #124883 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2.6-rc]
- #124862 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1.1-rc]
- #124861 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #124860 acceptance: TestCompose failed [KeyError: &#39;ContainerConfig&#39;] [C-test-failure O-robot T-dev-inf branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15542955?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15542955?buildTab=artifacts#/) on master @ [9bba582435b6fbeb70630b66415c496e4e93d230](https://github.com/cockroachdb/cockroach/commits/9bba582435b6fbeb70630b66415c496e4e93d230):


```
         Container gss_cockroach_1  Creating
         Container gss_cockroach_1  Created
         Container gss_psql_1  Creating
         Container gss_psql_1  Created
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun  6 15:08 ca.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun  6 15:08 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun  6 15:08 client.root.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun  6 15:08 client.root.key
        psql_1       | -rw------- 1 roach roach 1218 Jun  6 15:08 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2611 Jun  6 15:08 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun  6 15:08 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun  6 15:08 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1218 Jun  6 15:08 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun  6 15:08 node.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun  6 15:08 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=7028c0004bc0
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.32.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        Aborting on container exit...
        
�[Kpsql_1 exited with code 2
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (29.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125089 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15599945?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15599945?buildTab=artifacts#/) on master @ [41c37cbcc53d34b01d5070c52158f63bb78edd13](https://github.com/cockroachdb/cockroach/commits/41c37cbcc53d34b01d5070c52158f63bb78edd13):


```
         Container gss_cockroach_1  Creating
         Container gss_cockroach_1  Created
         Container gss_psql_1  Creating
         Container gss_psql_1  Created
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun 10 23:56 ca.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 10 23:56 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun 10 23:56 client.root.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 10 23:56 client.root.key
        psql_1       | -rw------- 1 roach roach 1219 Jun 10 23:56 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2611 Jun 10 23:56 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun 10 23:56 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 10 23:56 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1219 Jun 10 23:56 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun 10 23:56 node.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 10 23:56 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=ca102ca915ef
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.32.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        
�[Kpsql_1 exited with code 2
        Aborting on container exit...
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (29.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125413 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #125089 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15633207?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15633207?buildTab=artifacts#/) on master @ [f414f7c2fa09dc73ce463050afdfe34d6e2246da](https://github.com/cockroachdb/cockroach/commits/f414f7c2fa09dc73ce463050afdfe34d6e2246da):


```
         Container gss_cockroach_1  Creating
         Container gss_cockroach_1  Created
         Container gss_psql_1  Creating
         Container gss_psql_1  Created
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun 12 21:30 ca.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 12 21:30 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun 12 21:30 client.root.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 12 21:30 client.root.key
        psql_1       | -rw------- 1 roach roach 1218 Jun 12 21:30 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2611 Jun 12 21:30 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun 12 21:30 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 12 21:30 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1218 Jun 12 21:30 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun 12 21:30 node.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 12 21:30 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=c576ac1548fa
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.32.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        
�[Kpsql_1 exited with code 2
        Aborting on container exit...
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (31.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125413 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #125089 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15710324?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15710324?buildTab=artifacts#/) on master @ [d88af9d8d14c906b18a4247ddbdba5f56cba71c9](https://github.com/cockroachdb/cockroach/commits/d88af9d8d14c906b18a4247ddbdba5f56cba71c9):


```
         Container gss_cockroach_1  Creating
         Container gss_cockroach_1  Created
         Container gss_psql_1  Creating
         Container gss_psql_1  Created
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun 17 23:24 ca.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 17 23:24 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun 17 23:24 client.root.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 17 23:24 client.root.key
        psql_1       | -rw------- 1 roach roach 1217 Jun 17 23:24 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2627 Jun 17 23:24 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun 17 23:24 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 17 23:24 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1218 Jun 17 23:24 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun 17 23:24 node.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 17 23:24 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=a143d180ff94
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.32.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        Aborting on container exit...
        
�[Kpsql_1 exited with code 2
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (26.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125608 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2.6-rc]
- #125413 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #125089 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

acceptance.TestCompose [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15711004?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/15711004?buildTab=artifacts#/) on master @ [5e72cfccd672c1a66e56df12fe1191219c2df7d1](https://github.com/cockroachdb/cockroach/commits/5e72cfccd672c1a66e56df12fe1191219c2df7d1):


```
        Attaching to cockroach_1, kdc_1, psql_1
        kdc_1        | Authenticating as principal root/admin@MY.EX with password.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes256-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | Entry for principal postgres/gss_cockroach_1.gss_default@MY.EX with kvno 2, encryption type aes128-cts-hmac-sha1-96 added to keytab WRFILE:/keytab/crdb.keytab.
        kdc_1        | krb5kdc: starting...
        psql_1       | Available certs:
        psql_1       | total 44
        psql_1       | -rw-r--r-- 1 roach roach 1151 Jun 18 01:40 ca.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 18 01:40 ca.key
        psql_1       | -rw-r--r-- 1 roach roach 1237 Jun 18 01:40 client.root.crt
        psql_1       | -rw------- 1 roach roach 1679 Jun 18 01:40 client.root.key
        psql_1       | -rw------- 1 roach roach 1219 Jun 18 01:40 client.root.key.pk8
        psql_1       | -rw------- 1 roach roach 2611 Jun 18 01:40 client.root.pk12
        psql_1       | -rw-r--r-- 1 roach roach 1253 Jun 18 01:40 client.testuser.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 18 01:40 client.testuser.key
        psql_1       | -rw------- 1 roach roach 1216 Jun 18 01:40 client.testuser.key.pk8
        psql_1       | -rw-r--r-- 1 roach roach 1204 Jun 18 01:40 node.crt
        psql_1       | -rw------- 1 roach roach 1675 Jun 18 01:40 node.key
        psql_1       | Environment:
        psql_1       | PGPORT=26257
        psql_1       | HOSTNAME=786adf037d2c
        psql_1       | HOME=/home/roach
        psql_1       | PG_VERSION=15.4-1.pgdg120+1
        psql_1       | PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/15/bin
        psql_1       | LANG=en_US.utf8
        psql_1       | PG_MAJOR=15
        psql_1       | GOSU_VERSION=1.16
        psql_1       | PGHOST=cockroach
        psql_1       | PWD=/
        psql_1       | PGDATA=/var/lib/postgresql/data
        psql_1       | Creating a k5s token...
        psql_1       | Password for tester@MY.EX: 
        psql_1       | Preparing SQL user ahead of test
        psql_1       | psql: error: connection to server at "cockroach" (192.168.32.3), port 26257 failed: Connection refused
        psql_1       | 	Is the server running on that host and accepting TCP/IP connections?
        cockroach_1  | *
        cockroach_1  | * WARNING: Running a server without --sql-addr, with a combined RPC/SQL listener, is deprecated.
        cockroach_1  | * This feature will be removed in a later version of CockroachDB.
        cockroach_1  | *
        
�[Kpsql_1 exited with code 2
        Aborting on container exit...
         Container gss_psql_1  Stopping
         Container gss_psql_1  Stopped
         Container gss_cockroach_1  Stopping
         Container gss_cockroach_1  Stopped
         Container gss_kdc_1  Stopping
         Container gss_kdc_1  Stopped
        
    compose_test.go:100: exit status 2
    --- FAIL: TestCompose/ComposeGSS (32.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125608 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2.6-rc]
- #125413 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-24.1]
- #125089 acceptance: TestCompose failed [port 26257 failed: Connection refused] [C-test-failure O-robot T-dev-inf branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCompose.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/acceptance/compose/gss/docker-compose-python.yml
+++ b/pkg/acceptance/compose/gss/docker-compose-python.yml
@@ -17,11 +17,17 @@ services:
       - ${CERTS_DIR:-../../.localcluster.certs}:/certs
       - keytab:/keytab
       - ${COCKROACH_BINARY:-../../../../cockroach-linux-2.6.32-gnu-amd64}:/cockroach/cockroach
+    healthcheck:
+      test: "bash -c 'echo -n > /dev/tcp/cockroach/26257'"
+      interval: 0.5s
+      timeout: 10s
+      retries: 25
   python:
     image: us-east1-docker.pkg.dev/crl-ci-images/cockroach/acceptance-gss-python:20221214-141947
     user: "${UID}:${GID}"
     depends_on:
-      - cockroach
+      cockroach:
+        condition: service_healthy
     environment:
       - PGHOST=cockroach
       - PGPORT=26257

--- a/pkg/acceptance/compose/gss/docker-compose.yml
+++ b/pkg/acceptance/compose/gss/docker-compose.yml
@@ -1,3 +1,4 @@
+version: '3'
 services:
   kdc:
     image: us-east1-docker.pkg.dev/crl-ci-images/cockroach/acceptance-gss-kdc:20221214-131000
@@ -17,11 +18,17 @@ services:
       - ${CERTS_DIR:-../../.localcluster.certs}:/certs
       - keytab:/keytab
       - ${COCKROACH_BINARY:-../../../../cockroach-linux-2.6.32-gnu-amd64}:/cockroach/cockroach
+    healthcheck:
+      test: "bash -c 'echo -n > /dev/tcp/cockroach/26257'"
+      interval: 0.5s
+      timeout: 10s
+      retries: 25
   psql:
     image: us-east1-docker.pkg.dev/crl-ci-images/cockroach/acceptance-gss-psql:20230907-113902
     user: "${UID}:${GID}"
     depends_on:
-      - cockroach
+      cockroach:
+        condition: service_healthy
     environment:
       - PGHOST=cockroach
       - PGPORT=26257

